test/integration/roundtrip_test.go

```diff
@@ -0,0 +1,291 @@
+		}
+	}()
+
+	if err := waitForApiserverUp(serverIP, serverFailed); err != nil {
+		t.Fatalf("%v", err)
+	}
+
+	testSwaggerSpec(t, serverIP)
+	testAPIGroupList(t, serverIP)
+	testAPIGroup(t, serverIP)
+	testAPIResourceList(t, serverIP)
+
+	config := &restclient.Config{}
+	config.Host = "https://localhost:8888"
+	config.Insecure = true
+	//	config.TLSClientConfig = restclient.TLSClientConfig{
+	// CertFile: secureServingOptions.ServerCert.CertKey.CertFile,
+	// KeyFile:  secureServingOptions.ServerCert.CertKey.KeyFile,
+	// CAFile:  secureServingOptions.ServerCert.CACertFile,
+	//	}
+	clientset, err := servicecatalog.NewForConfig(config)
+	if nil != err {
+		t.Fatal("can't make the client from the config", err)
+	}
+	gv := clientset.Servicecatalog().RESTClient().APIVersion()
+	t.Log(gv)
+
+	broker, err := clientset.Servicecatalog().Brokers().Create(
+		&v1alpha1.Broker{ObjectMeta: v1.ObjectMeta{Name: "test-broker"},
+			Spec: v1alpha1.BrokerSpec{URL: "broker needs a url"}})
+	t.Log(broker, err)
+
+	brokers, err := clientset.Servicecatalog().Brokers().List(v1.ListOptions{})
+	t.Log(brokers, err)
+
+	brokerAgain, err := clientset.Servicecatalog().Brokers().Get("test-broker")
+	t.Log(brokerAgain, err)
+	brokerAgain.Status.Conditions = append(brokerAgain.Status.Conditions,
+		v1alpha1.BrokerCondition{
+			Type:   v1alpha1.BrokerConditionReady,
+			Status: v1alpha1.ConditionTrue,
+		},
+	)
+	brokerU := brokerAgain
+	t.Log(brokerU)
+	brokerAgain, err = clientset.Servicecatalog().Brokers().Update(brokerU)
+	t.Log("updated broker, status should not be different", brokerAgain, err)
+	brokerAgain, err = clientset.Servicecatalog().Brokers().Get("test-broker")
+	t.Log("updated broker, status should not be different", brokerAgain, err)
+
+	brokerAgain, err = clientset.Servicecatalog().Brokers().UpdateStatus(brokerU)
+	t.Log("updated broker status, status should be different", brokerAgain, err)
+	brokerAgain, err = clientset.Servicecatalog().Brokers().Get("test-broker")
+	t.Log("updated broker status, status should be different", brokerAgain, err)
+
+	err = clientset.Servicecatalog().Brokers().Delete("test-broker", &v1.DeleteOptions{})
+
+	brokerAgain, err = clientset.Servicecatalog().Brokers().Get("test-broker")
+	t.Log("updated broker status, status should be different", brokerAgain, err)
+}
+
+func waitForApiserverUp(serverIP string, stopCh <-chan struct{}) error {
+	minuteTimeout := time.After(time.Minute)
+	for {
+		select {
+		case <-stopCh:
+			return fmt.Errorf("apiserver failed")
+		case <-minuteTimeout:
+			return fmt.Errorf("waiting for apiserver timed out")
+		default:
+			glog.Errorf("Waiting for : %#v", serverIP)
+			tr := &http.Transport{
+				TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+			}
+			client := &http.Client{Transport: tr}
+			_, err := client.Get(serverIP)
+			if err == nil {
+				return nil
+			}
+		}
+		<-time.After(100 * time.Millisecond)
+	}
+}
+
+func readResponse(serverURL string) ([]byte, error) {
```

Comment:
everything that follows was a direct copy from the upstream example. Need to evaluate whether the tests are good or not.